src/DataTypes/DataTypeNumberBase.cpp

```diff
@@ -11,6 +11,55 @@ Field DataTypeNumberBase<T>::getDefault() const
 {
     return NearestFieldType<FieldType>();
 }
+template <typename T>
+String DataTypeNumberBase<T>::getSQLCompatibleName() const
+{
+    if constexpr (std::is_same_v<T, Int8>)
```

Comment:
(I don't care too much but parentheses for single-line if/for/while statements are usually omitted in the codebase. In this case, parentheses blow the code artificially up.)